dbt-bigquery/src/dbt/adapters/bigquery/python_submissions.py

```diff
@@ -163,3 +176,181 @@ def _update_batch_from_config(
+        try:
+            bucket_name = gcs_uri.split("gs://")[1].split("/")[0]
+            blob_name = "/".join(gcs_uri.split("gs://")[1].split("/")[1:])
+
+            bucket = self._storage_client.bucket(bucket_name)
+            blob = bucket.blob(blob_name)
+            file_content = blob.download_as_text()
+
+            data = json.loads(file_content)
+            return data
+
+        except json.JSONDecodeError:
+            _logger.error(f"Error: File '{gcs_uri}' content is not valid JSON.")
+            return None  # Or raise the exception if you prefer
+
+        except Exception as e:
+            _logger.exception(f"Error reading file from GCS: {e}")
+            return None
+
+    def _submit_bigframes_job(
+        self, notebook_compiled_code: str, notebook_template_id: str
+    ) -> None:
+
+        notebook_execution_job = aiplatform_v1beta1.NotebookExecutionJob()
+        notebook_execution_job.notebook_runtime_template_resource_name = f"projects/{self._credentials.execution_project}/locations/{self._credentials.bigframes_region}/notebookRuntimeTemplates/{notebook_template_id}"
+
+        if self._credentials.bigframes_upload_notebook_gcs:
+            notebook_execution_job.gcs_notebook_source = (
+                aiplatform_v1beta1.NotebookExecutionJob.GcsNotebookSource(
+                    uri=self._gcs_location
+                )
+            )
+        else:
+            # Exec directly from raw bytes
+            notebook_compiled_code_bytes = notebook_compiled_code.encode("utf-8")
+            notebook_execution_job.direct_notebook_source = (
+                aiplatform_v1beta1.NotebookExecutionJob.DirectNotebookSource(
+                    content=notebook_compiled_code_bytes
+                )
+            )
+
+        notebook_execution_job.service_account = (
+            self._GoogleCredentials._service_account_email
+        )
+        notebook_execution_job.gcs_output_uri = "gs://{}/{}/logs".format(
+            self._credentials.gcs_bucket, self._model_file_name
+        )
+        notebook_execution_job.display_name = self._get_batch_id()
+        request = aiplatform_v1beta1.CreateNotebookExecutionJobRequest(
+            parent=f"projects/{self._credentials.execution_project}/locations/{self._credentials.bigframes_region}",
+            notebook_execution_job=notebook_execution_job,
+        )
+
+        res = self._ai_platform_client.create_notebook_execution_job(
+            request=request
+        ).result()
+
+        job_id = self._get_job_id(res.name)
+        gcs_log_uri = (
+            f"{notebook_execution_job.gcs_output_uri}/{job_id}/{self._model_name}.py"
+        )
+        gcs_log = self._read_json_from_gcs(gcs_log_uri)
+        # TODO(jialuo): improve the logger info.
+        _logger.info(
+            f"The colab notebook runtime outputs from GCS: {gcs_log['cells'][0]['outputs']}"
+        )
+
+        _ = self._ai_platform_client.get_notebook_execution_job(name=res.name)
```

Comment:
Is there a default polling retry on this? I would have expected that we use `self._result_polling_policy` here. Looking back, I don't see where else that gets used.

Separately, we don't need to specify `_ =` here if we're not doing anything with the result.